### ISSUE TITLE
Restrict synchronisation to data that has been changed in a specified time period [3]

### ISSUE THREAD

**Issue by MateStrysewske** [hzi-braunschweig/SORMAS-Project#7305]
### Situation Description

The decision which data is synchronised to mobile devices currently depends on whether or not the data is archived. Automatic archiving is by default done after 90 days of inactivity, but can be adjusted. In situations where a lot of data is collected, this can result in too much data being pushed to the mobile devices.

### Feature Description

* Add a new Integer property called `MAX_CHANGE_DATE_PERIOD` to the `REDUCED_SYNCHRONIZATION` feature introduced by #7304; default value null
* If the feature configuration is enabled and the new property != null, when pulling production data (i.e. non-infrastructure data) from the web app, exclude data with `changeDate < today - MAX_CHANGE_DATE_PERIOD` (make sure to use the change date filter here that includes child entities)
* If the property is enabled, after a successful synchronization, delete all data from the mobile app where `changeDate < today - MAX_CHANGE_DATE_PERIOD`

**Comment by bernardsilenou** [comment]
@MateStrysewske 
- This may not be very important for this issue for now, can we have a solution that can easily be extended to depend on the  disease followup period. Reason is that as we add more diseases in sormas, It would be difficult to find a value that matches all the diseases.
- To use something like: last change date == followup  duration (d), or 2d, or nd. 
- The d differs for different diseases, 14 days for covid-1 and  even 90 - 365 days for other diseases

**Comment by MateStrysewske** [comment]
@bernardsilenou Could be a good idea, but definitely for a follow-up issue

**Comment by MateStrysewske** [comment]
I'd recommend to not start with this issue until #9437 is merged because there's a risk for significant merge conflicts otherwise.

**Comment by markusmann-vg** [comment]
https://github.com/hzi-braunschweig/SORMAS-Project/issues/9437 is done so we can pick it up. I moved it to backlog again

**Comment by andrei7bal** [comment]
Hello @MateStrysewske !

Is this feature suppose to be added just for the cases entities right? Currently the _REDUCED_SYNCHRONIZATION_ feature is added just for cases.

Thanks

**Comment by MateStrysewske** [comment]
@andrei7bal Hey! No, this is supposed to be applied to all "production" data, i.e. stuff like cases, contacts, visits, events, etc., but not infrastructure data and also not meta data like feature configurations, disease configurations, etc.

Based on your comment it probably makes sense to refactor the REDUCED_SYNCHRONIZATION feature though, i.e. move the logic that is currently connected to it to a new property called `LIMIT_NOT_A_CASE_SYNC` so that turning on REDUCED_SYNCHRONIZATION without turning on one of the properties essentially doesn't do anything.

**Comment by StefanKock** [comment]
Decided in the team daily to move to next version (not merge directly before release)

**Comment by StefanKock** [comment]
It was decided on 2022-12-16 to not merge this ticket into 1.78.x because it was not finished in-time.

**Comment by MateStrysewske** [comment]
Testing info for QA: This feature can't really be properly tested on a test instance because it requires the mobile app to do an "obsolete sync" which is only done once per day, so I suggest to test it locally with Android Studio up and running.

Basically, what is supposed to happen is that data that has not been changed in X days should not be synchronized TO the mobile app and removed FROM the mobile app; the amount of days is specified in the featureconfiguration table as the property MAX_CHANGE_DATE_PERIOD in the LIMITED_SYNCHRONIZATION feature. The default is -1, which basically turns the feature off.

The way I tested it were as follows:
- Set the value to -1, install the mobile app, log in, synchronize, verify that all data is on the device
- Set the value to 1, re-install the mobile app, log in, synchronize, verify that only data that has been changed since yesterday is on the device
- Change line 167 in the SynchronizeDataAsync class from `if (pullAndRemoveObsoleteNecessary) {` to `if (true) {` in order to always delete obsolete data with every synchronization
- Set the value to -1 again, install the mobile app, log in, synchronize, verify that all data is on the device
- Set the value to 1 and the change date to the current date, trigger a synchronization in the mobile app and verify that all data that has not been changed since yesterday is gone from the device
